charts/gitops-server/templates/deployment.yaml

```diff
@@ -36,6 +40,21 @@ spec:
             - "{{ .Release.Namespace }}"
             - "--log-level"
             - "{{ .Values.logLevel }}"
+            {{- if .Values.serverTLS.useSecret }}
+            - "--tls-cert-file"
+            - "/etc/tls-volume/{{ .Values.serverTLS.certContentField }}"
+            - "--tls-private-key-file"
+            - "/etc/tls-volume/{{ .Values.serverTLS.keyContentField }}"
+            {{- end }}
+            {{- if .Values.serverTLS.useLocalFiles  }}
+            - "--tls-cert-file"
+            - "/etc/tls-volume/.cert-file"
+            - "--tls-private-key-file"
+            - "/etc/tls-volume/.key-file"
+            {{- end }}
+            {{- if (not (and .Values.serverTLS.useLocalFiles .Values.serverTLS.useSecret) )}}
+            - "--insecure"
+            {{- end }}
```

Comment:
I feel this goes against most cert/sec UX patterns. With this logic, if they forget to set TLS files, it will quietly default to insecure. I believe that the majority of tools will default to secure (and complain if you don't have certs) and require explicitly mentioning that you want it to be insecure. (`curl`'s `-k` flag comes to mind). WDYT @SamLR?

Reply:
Fair point. I'll not default to `insecure` given the reasons you give.